enhancements/microshift/default-csi-plugin.md

```diff
@@ -0,0 +1,668 @@
+Single Node OpenShift (SNO). The plugin is compatible with MicroShift's recommended OS's(RHEL/Centos/Fedora).
+
+## Motivation
+<!---
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+--->
+
+For proof of concept, the KubeVirt Hostpath Provisioner (HPP) was chosen smooth the ramp-up of MicroShift development
+but was not intended for use in production. MicroShift has since matured enough to warrant a storage provisioner that
+will work out of the box and be compatible with a standard RHEL 8.X / RHEL for Edge host.
+
+ODF-LVM is production-ready, enables isolation of workload and system storage, and is already supported by ODF on Single
+Node OpenShift.  ODF-LVM implements features not present in HPP, such as volume snapshotting and thin-provisioning.
+
+### User Stories
+
+<!---
+Detail the things that people will be able to do if this is
+implemented. In each story, explain who the actor is based on their
+role, explain what they want to do with the system, and explain the
+underlying goal they have. Use the standard three part formula:
+
+"As a <role>, I want to <take some action> so that I can <accomplish a
+goal>."
+
+Make the change feel real for users, without getting bogged down in
+implementation details.
+
+Include a story on how this proposal will be operationalized:
+life-cycled, monitored and remediated at scale.
+--->
+
+* As an edge device owner, I want to image devices en masse with workload and storage manifests so that when the devices
+  are deployed to the field, persistent storage will be created automatically, prior to the workload starting.
+* As a device owner, I must be able to protect the host OS, or other workloads, from workloads that may over-consume
+  storage, which could destabilize the platform.
+* As a device owner, I want to configure system and cluster storage parameters pre-boot and have those configurations
+  acted upon when MicroShift starts.
+* As a device owner, I want to update cluster storage configuration with the assurance that a faulty config will not
+  destabilize the system.
+
+### Goals
+
+<!---
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+--->
+
+* Provide a default, production-ready, dynamic storage solution out of the box.
+* Protect the host OS from greedy workloads to assure system stability.
+* Align with MicroShift's lifecycle management model via rpm-ostree atomic updates.
+
+### Non-Goals
+
+<!---
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+--->
+
+* Support multi-node cluster topologies
+* Enable on-disk encryption
```

Comment:
We do want to support disk encryption, just via LUKS for whole disk and not separately through the CSI plugin for individual PVs. I'm not sure how best to rephrase this to capture that.

Reply:
In that case, we can probably just omit it entirely as it's not usually in scope for CSI.